quicvarint/varint.go

```diff
@@ -16,9 +14,9 @@ const (
 	maxVarInt8 = 4611686018427387903
 )
 
-// Read reads a number in the QUIC varint format
-func Read(b io.ByteReader) (uint64, error) {
-	firstByte, err := b.ReadByte()
+// Read reads a number in the QUIC varint format from r.
+func Read(r Reader) (uint64, error) {
```

Comment:
What do we get out of changing the `io.ByteReader` to a `Reader` here? `io.ByteReader` is the smaller interface, so it would be preferable to use that one instead of the `Reader`, wouldn't it?

Reply:
I did this mostly for symmetry, and to mimic the `http3.byteReader` interface (https://github.com/lucas-clemente/quic-go/blob/master/http3/frames.go#L13-L16). But I see your point!

Curious: did you ever benchmark an implementation of `Read` that uses `io.Reader` instead of `io.ByteReader`?

Reply:
> Curious: did you ever benchmark an implementation of `Read` that uses `io.Reader` instead of `io.ByteReader`?

I went ahead and tried this. An implementation of `Read` that uses `io.Reader` is roughly twice as slow as the existing `ReadByte`-based implementation.

Reply:
@marten-seemann OK, so I’m happy to either revert `Read` to accept an `io.ByteReader`, or keep it as a `quicvarint.Reader`.

Reply:
The Go philosophy says to use the smallest interface possible, so let's keep it an `io.ByteReader`.